trpc/queries/goals.ts

```diff
@@ -281,12 +287,21 @@ export const goalsFilter = (
             ...ownerFilter,
             ...participantFilter,
             ...projectFilter,
+            ...criteriaFilter,
             ...extra,
         },
         orderBy: orderBy.length ? orderBy : defaultOrderBy,
     };
 };
 
+export const getGoalActivityFilterIdsQuery = (): Prisma.GoalAchieveCriteriaFindManyArgs => ({
+    where: {
+        criteriaGoalId: { not: null },
+        deleted: { not: true },
+    },
+    select: { id: true },
```

Comment:
really need a `id` field instead of `criteriaGoalId`?

However you get `criteriaGoalId` in kysely query